casepro/msgs/models.py

```diff
@@ -384,14 +391,19 @@ def search(cls, org, user, search):
         if folder == MessageFolder.flagged:
             queryset = queryset.filter(is_flagged=True)
 
-        # archived messages can be implicitly or explicitly included depending on folder
-        if folder == MessageFolder.archived:
-            queryset = queryset.filter(is_archived=True)
-        elif folder == MessageFolder.flagged:
-            if not include_archived:
-                queryset = queryset.filter(is_archived=False)
+        # if this is a refresh we want everything with new actions
+        if last_refresh:
```

Comment:
If a user has the flagged view open, and another user unflags a message, it won't be removed in the first user's view - because you're only fetching flagged messages you can't see an updated message which has been unflagged.